extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js

```diff
@@ -1709,6 +1709,16 @@
 				value,
 				title;
 
+			if(data.TYPE === "SITEMAP_CHANGED") {
```

Comment:
isn't it also necessary to check if it is _this_ sitemap which changed? I.e. have you tried having to browser windows open with different sitemaps and check that only the affected one gets redirected to its main page?